homeassistant/components/lg_soundbar/media_player.py

```diff
@@ -139,16 +139,19 @@ def state(self):
     @property
     def sound_mode(self):
         """Return the current sound mode."""
-
         if self._equaliser == -1:
             return ""
+        if self._equaliser >= len(temescal.equalisers):
+            return ""
         return temescal.equalisers[self._equaliser]
 
     @property
     def sound_mode_list(self):
         """Return the available sound modes."""
         modes = []
         for equaliser in self._equalisers:
+            if equaliser >= len(temescal.equalisers):
+                temescal.equalisers.append("unknown " + str(equaliser))
```

Comment:
Entity state properties should not have side effects. Please move this to eg `update` method.

Reply:
> This looks like it should be fixed in the library instead.

I agree that it would be the optimum that it´s handeled by the library.
But it´s not the case and changing that would require changes in the lib as well as in this component.

This hacky approch is for sure not the optimum, but I´m as a user prefer to have a equaliser setting named e.g.: "unknown 18" instead of not beeing able to select the setting at all. 
Because the side effect is also that if I select the unknown setting using the lg app or the soundbars remote, in the lovelace dashboard I don´t see any selected sound mode.

I made a PR in the librarys repro to fix the missing sound mode. If that get´s accepted I would do another PR to there to handle that problem in the lib it´s self. It just seems kind if inactive to me and I´m not expecting too much.
Sorry writing half a book in this comment.